### PR TITLE
Fix the manifest path related to STATIC_ROOT

### PR DESCRIPTION
**Pull request by batisteo**
I think we shouldn’t use `urljoin` with STATIC_ROOT, as it is not an URL:

It could be defined as `/home/user/some-project/static`, and _urljoin_ will replace the “file” (last part of the _URL_) with the second argument, here the Django Vite URL prefix, eg. `build`.

So in this case, DJANGO_VITE_MANIFEST_PATH would be `/home/user/some-project/build` and is not what’s expected.

---
Also, we’d rely on DJANGO_VITE_DEV_MODE instead of `settings.DEBUG`.

We’re not always running the Vite server while developing our Django project, we’re checking the post usage in the settings.

Could you have a quick look @gnuletik?

### CODE REVIEW

File: django_vite/templatetags/django_vite.py [hunk 1]
```diff
@@ -38,19 +38,17 @@
 DJANGO_VITE_ASSETS_PATH = getattr(settings, "DJANGO_VITE_ASSETS_PATH")
 
 # Prefix for STATIC_URL
-DJANGO_VITE_STATIC_URL_PREFIX = getattr(
-    settings, "DJANGO_VITE_STATIC_URL_PREFIX", ""
-)
+DJANGO_VITE_STATIC_URL_PREFIX = getattr(settings, "DJANGO_VITE_STATIC_URL_PREFIX", "")
```
Comment (batisteo) [inline django_vite/templatetags/django_vite.py:41]: This is just _Black_ formatting. I could remove it if it bothers you.

**Comment by gnuletik** [comment]
Thanks for the PR @batisteo !

You're right, using `urljoin` generates a bad file path when `STATIC_ROOT` is defined without trailing slash.
Using  `path_join` fixes it :) LGTM.


Regarding `DJANGO_VITE_DEV_MODE` instead of `settings.DEBUG`, I agree with you.
@MrBin99 [introduced it](https://github.com/MrBin99/django-vite/commit/7f7c010191e2b923ab3e4fd1ab9991d2c3f98e3a#diff-340facb9cc1393cad64ce6b5408114ea99338657f8386e4a0d99e97c4b01e2d8R42-R49) so I'm not sure if there is a usecase here.

File: django_vite/templatetags/django_vite.py (hunk 1 above)
Comment (MrBin99) [reply django_vite/templatetags/django_vite.py:41]: No thanks !
I plan to introduce Black and Flake8 with pre-commit and Github Actions as there is more people contributing to the project.

**Comment by MrBin99** [comment]
I agree with both of you.
I think we need to support `pathlib.Path` like mentionned here #28. It will be easier to mange paths.
For `DJANGO_VITE_DEV_MODE`, I agree we should aways rely on this variable and directly with Django `DEBUG`.

I will try to work on the `Pathlib` support and formatting by the end of this week (if I can find time to do so 😊).

Thanks for your work !

**Comment by batisteo** [comment]
@MrBin99 Do you think you could do a release?
I’m afraid switching to `pathlib` (that I encourage!) might impose the need to right tests and delay a release.

**Comment by MrBin99** [comment]
@batisteo Yeah I can do a release before switching and maybe bump the version to 2.0 after because, I agree, migrating to pathlib will break a lot of code. So I bump the version and people will be aware 1.X no pathlib 2.X pathlib.
What do you think ?

**Comment by batisteo** [comment]
A tiny `1.3.1` would do fine for me!

**Comment by MrBin99** [comment]
Too late 😂

**Pull request merged**